src/register/RegistrationFields/UsernameField/UsernameField.jsx

```diff
@@ -91,7 +91,7 @@ const UsernameField = (props) => {
     event.preventDefault();
     handleErrorChange('username', ''); // clear error
     handleChange({ target: { name: 'username', value: suggestion } }); // to set suggestion as value
-    dispatch(clearUsernameSuggestions());
+    dispatch(clearUsernameSuggestions(true));
```

Comment:
why do we need to set true here?

Reply:
Username suggestions are cleared when either the user clicks a suggestion of clicks clear button. We want to backup suggestions only when user clicks a suggestion therefore i introduced a temporary argument `isSuggestionClicked` in `clearUsernameSuggestion` and pass true only when the user clicked the suggestion. So that we can backup it.